torch/backends/_nnapi/prepare.py

```diff
@@ -0,0 +1,167 @@
+from typing import Optional, List, Tuple, Any
+
+import torch
+from torch.backends._nnapi.serializer import serialize_model
+
+class NnapiModule(torch.nn.Module):
+    """Torch Module that wraps an NNAPI Compilation.
+
+    This module handles preparing the weights, initializing the
+    NNAPI TorchBind object, and adjusting the memory formats
+    of all inputs and outputs.
+    """
+
+    comp: Optional[torch.classes._nnapi.Compilation]
+
+    def __init__(
+            self,
+            ser_model: torch.Tensor,
+            weights: List[torch.Tensor],
+            inp_mem_fmts: List[int],
+            out_mem_fmts: List[int],
+            out_templates: List[torch.Tensor]):
+        super().__init__()
+        self.ser_model = ser_model
```

Comment:
@dreiss Okay I think I now understand why you did this. Your serialized model (for NNAPI) is just a Tensor<int8> represented as a regular member of the original model so that it will get naturally serialized when the model is serialized to torchscript. Is that right?

Reply:
Yes.